sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferStore.scala

```diff
@@ -460,3 +456,16 @@ abstract class RapidsBufferStore(val tier: StorageTier)
     override def toString: String = s"$name buffer size=${getMemoryUsedBytes}"
   }
 }
+
+/**
+ * Buffers that inherit from this type do not support changing the spillable status
+ * of a `RapidsBuffer`. This is only used right now for disk and GDS.
+ * @param tier storage tier of this store
+ */
+abstract class RapidsBufferStoreWithoutSpillabilitySupport(override val tier: StorageTier)
```

Comment:
Super minor name suggestion for brevity, not must-fix
```suggestion
abstract class RapidsBufferStoreWithoutSpill(override val tier: StorageTier)
```

Reply:
oh that's better, will do.